src/pages/ProviderEServiceCreatePage/components/EServiceCreateStepGeneral/EServiceCreateStepGeneral.tsx

```diff
@@ -342,3 +328,31 @@ export const EServiceCreateStepGeneralSkeleton: React.FC = () => {
     </>
   )
 }
+
+function evaluateFormDefaultValues(
+  isEServiceFromTemplate: boolean,
+  template: EServiceTemplateDetails | undefined,
+  descriptor: ProducerEServiceDescriptor | undefined,
+  eserviceMode: EServiceMode
+): EServiceCreateStepGeneralFormValues {
+  if (isEServiceFromTemplate)
+    return {
+      name: descriptor?.eservice.name ?? '',
+      description: descriptor?.eservice.description ?? '',
+      technology: descriptor?.eservice.technology ?? 'REST',
+      mode: eserviceMode,
+      isSignalHubEnabled: descriptor?.eservice.isSignalHubEnabled ?? false,
+      isConsumerDelegable: descriptor?.eservice.isConsumerDelegable ?? false,
+      isClientAccessDelegable: descriptor?.eservice.isClientAccessDelegable ?? false,
+    }
+
+  return {
+    name: template?.name ?? '',
```

Comment:
template should not be optional here, if we arrive here and it is `undefined` something is wrong with the rest of the logic